core/src/main/java/io/questdb/std/BoolList.java

```diff
@@ -157,6 +157,17 @@ public void removeIndex(int index) {
 
     public boolean replace(int index, boolean value) {
         assert index < pos;
+        return replace0(index, value);
+    }
+
+    public boolean extendAndReplace(int index, boolean value) {
+        if (index >= pos) {
+            setPos(index + 1);
```

Comment:
If you get inside this if it should always return `NO_ENTRY_VALUE` but it won't always happen.

I believe something like this will fail

```
list = new BoolList();
list.setAll(1, false);
set.extendAndReplace(2, true);
list.setAll(1, false);
Assert.isFalse(set.extendAndReplace(2, true));
```

Reply:
good catch, fixed it